eclipse-che/che-server
# chore: Apply git user data configmap to mount to /etc/gitconfig

### What does this PR do?
Add a workspace provisioner that checks registered git oAuth providers. When first provider found, a configmap with git user data, fetched from the corresponding git service, with mount to `/etc/gitconfig` is created.
 
### Screenshot/screencast of this PR



### What issues does this PR fix or reference?

https://github.com/eclipse/che/issues/20938

### How to test this PR?

1. Deploy che to a cluster: `chectl server:deploy --platform=openshift -i=quay.io/ivinokur/che-server:next`
where `quay.io/ivinokur/che-server:next` is `che-server` image built from the PR.
2. [Configure GitHub oAth](https://www.eclipse.org/che/docs/che-7/administration-guide/configuring-authorization/#configuring-github-oauth_che)
3. Start a factory from a private GitHub repo e.g. `https://github.com/vinokurig/test-extension/tree/devfile2`
4. Check that `workspace-userdata-gitconfig` configmap is created in the user namespace:
```
kind: ConfigMap
apiVersion: v1
metadata:
  name: workspace-userdata-gitconfig
  namespace: <user namespace>
  labels:
    controller.devfile.io/mount-to-devworkspace: 'true'
    controller.devfile.io/watch-configmap: 'true'
  annotations:
    controller.devfile.io/mount-as: subpath
    controller.devfile.io/mount-path: /etc/
data:
  gitconfig: "[user]\n\tname = <username>\n\temail = <email>"
```
Tested with `github.com`. Didn't test with gitlab and bitbucket as only on-premises versions are supported.

### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)


### Reviewers

Reviewers, please comment how you tested the PR when approving it.
